src/types/redis_bloom_chain.cc

```diff
@@ -147,18 +171,22 @@ rocksdb::Status BloomChain::Add(const Slice &user_key, const Slice &item, int *r
 
   // insert
   if (!exist) {
-    if (metadata.size + 1 > metadata.GetCapacity()) {  // TODO: scaling would be supported later
-      return rocksdb::Status::Aborted("filter is full");
+    if (metadata.size + 1 > metadata.GetCapacity()) {
+      if (metadata.IsScaling()) {
+        s = createBloomFilter(ns_key, &metadata);
```

Comment:
the status is not checked

Reply:
I'm not sure it can be done like this. Now we have holding a "lock" here. And `createBloomFilter` will directly call `Write` finally

I propose a `createBloomFilterInBatch` or other, the interface would just put the new bloomFilter in batch. The previous create can make use of this, and here we can just put it in batch first?